thetis/equation.py

```diff
@@ -0,0 +1,153 @@
+"""
+Implements Equation and Term classes.
+
+"""
+from utility import *
+
+
+class Term(object):
+    """
+    Implements a single term of an equation.
+    """
+    def __init__(self, function_space, boundary_markers, boundary_len):
+        # define bunch of members needed to construct forms
+        self.function_space = function_space
+        self.mesh = self.function_space.mesh()
+        self.test = TestFunction(self.function_space)
+        self.tri = TrialFunction(self.function_space)
+        self.normal = FacetNormal(self.mesh)
+        # TODO construct them here from mesh ?
+        self.boundary_markers = boundary_markers
```

Comment:
You can get the complete set of boundary markers on the mesh from: `mesh.exterior_facets.unique_markers`.